### ISSUE TITLE
2.0.0 crashes on startup (no network state permission?)

### ISSUE THREAD

**Issue by scy** [bitwarden/mobile#536]
Congratulations to your 2.0 release! I've installed it right away via F-Droid—and it doesn't work. 😁

I had a previous installation, starting the new version caused it to instantly crash. So I've cleared all application data, maybe it had problems with the data of the 1.x version.

Indeed, the app started up and showed me a login prompt. Once I filled that though and tapped "log in", it crashed again, and now crashes again on every startup.

I've hooked up my phone to my laptop and ran `adb logcat`, and when crashing, the following shows up:

```
06-13 18:03:01.010  1406  1406 I MonoDroid: UNHANDLED EXCEPTION:
06-13 18:03:01.043  1406  1429 I zygote  : Explicit concurrent copying GC freed 3729(1180KB) AllocSpace objects, 2(40KB) LOS objects, 55% free, 1213KB/2MB, paused 110us total 8.529ms
06-13 18:03:01.054  1406  1406 I MonoDroid: Xamarin.Essentials.PermissionException: You need to declare the permission: `android.permission.ACCESS_NETWORK_STATE` in your AndroidManifest.xml
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Xamarin.Essentials.Permissions.PlatformEnsureDeclared (Xamarin.Essentials.PermissionType permission, System.Boolean throwIfMissing) [0x00092] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Xamarin.Essentials.Permissions.EnsureDeclared (Xamarin.Essentials.PermissionType permission) [0x00000] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Xamarin.Essentials.Connectivity.get_PlatformNetworkAccess () [0x00000] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Xamarin.Essentials.Connectivity.get_NetworkAccess () [0x00000] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Bit.App.App.SyncIfNeeded () [0x0000d] in <a0a7c1dc68e14ae8bcc524b4a83fd6df>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Bit.App.App+<OnStart>d__22.MoveNext () [0x0014c] in <a0a7c1dc68e14ae8bcc524b4a83fd6df>:0
06-13 18:03:01.054  1406  1406 I MonoDroid: --- End of stack trace from previous location where exception was thrown ---
06-13 18:03:01.054  1406  1406 I MonoDroid:   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in <e35b06a4cfb64e2ea788a61ca9054dab>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <89e21e1747bb45cc99a01fd7e34945db>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <89e21e1747bb45cc99a01fd7e34945db>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <89e21e1747bb45cc99a01fd7e34945db>:0
06-13 18:03:01.054  1406  1406 I MonoDroid:   at (wrapper dynamic-method) System.Object.25(intptr,intptr)
06-13 18:03:01.061  1406  1406 W zygote  : JNI RegisterNativeMethods: attempt to register 0 native methods for android.runtime.JavaProxyThrowable
06-13 18:03:01.064  1406  1406 D AndroidRuntime: Shutting down VM
06-13 18:03:01.064  1406  1406 E AndroidRuntime: FATAL EXCEPTION: main
06-13 18:03:01.064  1406  1406 E AndroidRuntime: Process: com.x8bit.bitwarden, PID: 1406
06-13 18:03:01.064  1406  1406 E AndroidRuntime: android.runtime.JavaProxyThrowable: Xamarin.Essentials.PermissionException: You need to declare the permission: `android.permission.ACCESS_NETWORK_STATE` in your AndroidManifest.xml
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Xamarin.Essentials.Permissions.PlatformEnsureDeclared (Xamarin.Essentials.PermissionType permission, System.Boolean throwIfMissing) [0x00092] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Xamarin.Essentials.Permissions.EnsureDeclared (Xamarin.Essentials.PermissionType permission) [0x00000] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Xamarin.Essentials.Connectivity.get_PlatformNetworkAccess () [0x00000] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Xamarin.Essentials.Connectivity.get_NetworkAccess () [0x00000] in <1e18a28d6fc94816b5f83f6695c62676>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Bit.App.App.SyncIfNeeded () [0x0000d] in <a0a7c1dc68e14ae8bcc524b4a83fd6df>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Bit.App.App+<OnStart>d__22.MoveNext () [0x0014c] in <a0a7c1dc68e14ae8bcc524b4a83fd6df>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime: --- End of stack trace from previous location where exception was thrown ---
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in <e35b06a4cfb64e2ea788a61ca9054dab>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <89e21e1747bb45cc99a01fd7e34945db>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <89e21e1747bb45cc99a01fd7e34945db>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <89e21e1747bb45cc99a01fd7e34945db>:0
06-13 18:03:01.064  1406  1406 E AndroidRuntime:   at (wrapper dynamic-method) System.Object.25(intptr,intptr)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at mono.java.lang.RunnableImplementor.n_run(Native Method)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:790)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6494)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
06-13 18:03:01.064  1406  1406 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
06-13 18:03:01.071  3648  6054 W ActivityManager:   Force finishing activity com.x8bit.bitwarden/.MainActivity
06-13 18:03:01.091  3648  3705 I ActivityManager: Showing crash dialog for package com.x8bit.bitwarden u0
```

The important part seems to be `You need to declare the permission: `android.permission.ACCESS_NETWORK_STATE` in your AndroidManifest.xml`.

Sorry for bringing bad news, better luck next time! 😉

**Comment by scy** [comment]
Ah, 28166f79a158d2d1d13afec6d3b0490af9b5d7ee looks like you're already on it 😉👍

**Comment by kspearrin** [comment]
For whatever reason, the FDroid build strips the permission from the manifest. I don't know why, but I have explicitly added it now. Should be fixed in next v2.0.1.
